contracts/KyberPoolMaster.sol

```diff
@@ -260,6 +264,112 @@ contract KyberPoolMaster is Ownable {
         return getEpochDFeeData(curEpoch);
     }
 
+    /**
+     * @dev  Queries the amount of unclaimed rewards for the pool
+     *       return 0 if PoolMaster has calledRewardMaster
+     *       return 0 if staker's reward percentage in precision for the epoch is 0
+     *       return 0 if total reward for the epoch is 0
+     * @param epoch for which epoch is querying unclaimed reward
+     */
+    function getUnclaimedRewards(uint256 epoch) public view returns (uint256) {
+        if (claimedPoolReward[epoch]) {
+            return 0;
+        }
+
+        uint256 perInPrecision = kyberDAO.getStakerRewardPercentageInPrecision(
+            address(this),
+            epoch
+        );
+        if (perInPrecision == 0) {
+            return 0;
+        }
+
+        uint256 rewardsPerEpoch = kyberFeeHandler.rewardsPerEpoch(epoch);
+        if (rewardsPerEpoch == 0) {
+            return 0;
+        }
+
+        uint256 unclaimed = rewardsPerEpoch.mul(perInPrecision).div(PRECISION);
+
+        return unclaimed;
+    }
+
+    /**
+     * @dev  Claims rewards and distribute fees and its share to poolMaster
+     * @param epoch for which rewards are being claimed
+     */
+    function claimRewardsMaster(uint256 epoch) public {
+        require(!claimedPoolReward[epoch], "cRMaster: rewards already claimed");
+
+        uint256 unclaimed = getUnclaimedRewards(epoch);
+
+        require(unclaimed > 0, "cRMaster: no rewards to claim");
+
+        uint256 initialBalance = address(this).balance;
+
+        kyberDAO.claimReward(address(this), epoch);
+
+        uint256 totalRewards = address(this).balance.sub(initialBalance);
+
+        // TODO - define if this check is really necessary
+        require(
+            totalRewards >= unclaimed,
+            "cRMaster: claimed reward lower than expected"
+        );
```

Comment:
As discussed, this check doesn't provide any functionality or security to our contract, since we perform the calculation on the actually received rewards and not on what we expect to receive. Also, it will prevent the pool members and master to get their rewards if KyberDAO internally changes how to calculate the expected rewards. So let's remove it.